### PR TITLE
feat(breadcrumb): add template option for items

### LINKED ISSUE #618: Breadcrumb: Need feature for custom component and threshold
Hi Team,

We have two ways to create breadcrumbs using carbon breadcrumb component:

1. 1st way is, by writing breadcrumb-item in which we can pass our own ng-content. However, this method **doesn't support threshold**.
2. 2nd way is, without breadcrumb-item in which we pass items property as array of content and href. This way does support threshold but **don't allow us to send our custom ng-content**.

In our project, we want both threshold and custom template support.

Probable Solution- 
As per my finding carbon team will have to fix it in the below code in <ibm-breadcrumb> component in both showforoverflow if conditions:

Existing code is:
<ibm-breadcrumb-item *ngFor="let item of items" [href]="item.href">
 {{item.content}}
</ibm-breadcrumb-item>

changes needed:
You first need to check instance of item.content. If it is string then you can render as it is. However, if it is templateRef, you need to render it inside ng-container templateOutlet. So, user will be able to send custom template in content property. (something similar is already implemented in tooltip component)

(I am using below version of carbon:
"carbon-components": "^10.2.0",
"carbon-components-angular": "^3.7.0".)

@zvonimirfras

### PR DESCRIPTION
**Pull request by cal-smith**
closes #618

### CODE REVIEW

File: src/breadcrumb/breadcrumb-item.interface.ts [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
 	 */
 	content: string;
 	/**
 	 * Href for the breadcrumb item.
-	 * @type {string}
-	 * @memberof BreadcrumbItem
 	 */
 	href: string;
+	/**
+	 * Optional `TemplateRef` for the breadcrumb item. Recvices the content as an `$implicit` template variable
```
Comment (youda97) [inline src/breadcrumb/breadcrumb-item.interface.ts]:
```suggestion
	 * Optional `TemplateRef` for the breadcrumb item. Receives the content as an `$implicit` template variable
```

File: src/breadcrumb/breadcrumb.stories.ts [hunk 2]
```diff
@@ -59,6 +61,25 @@ storiesOf("Breadcrumb", module)
 		items: createBreadcrumbItems
 	}
 }))
+.add("Model with templates", () => ({
+	template: `
+	<ng-template #breadcrumbTemplate let-item>
+		Hello {{ item.content }}
```
Comment (youda97) [inline src/breadcrumb/breadcrumb.stories.ts]: Maybe make this a knob that allows them to change what's in the template

Comment (cal-smith) [reply src/breadcrumb/breadcrumb.stories.ts]: Yeah... could also make the `item.content` configurable as well 🤔

File: src/datepicker/datepicker.stories.ts [hunk 3]
```diff
@@ -260,7 +260,7 @@ storiesOf("Date Picker", module)
 			date: new Date(new Date().getFullYear(), 5, 15),
 			rangeDates: [
 				new Date(new Date().getFullYear(), 5, 15),
-				new Date(new Date().getFullYear(), 9, 19)
+				new Date(new Date().getFullYear(), 8, 19)
```
Comment (youda97) [inline src/datepicker/datepicker.stories.ts]: Not sure if this was intentional or not 😅

Comment (cal-smith) [reply src/datepicker/datepicker.stories.ts]: I was off by a year before 😏 so yeah, intentional ... just an easter egg

**Pull request merged**